src/Common/Notifications/Notifications.php

```diff
@@ -0,0 +1,347 @@
+<?php
+/**
+ * Handles In-App Notifications setup and actions.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Common\Notifications
+ */
+
+namespace TEC\Common\Notifications;
+
+use Tribe__Main;
+use TEC\Common\Admin\Conditional_Content\Dismissible_Trait;
+
+/**
+ * Class Notifications
+ *
+ * @since   TBD
+
+ * @package TEC\Common\Notifications
+ */
+final class Notifications {
+	use Dismissible_Trait;
+
+	/**
+	 * The slugs for plugins that support In-App Notifications.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	private $slugs = [];
+
+	/**
+	 * The Notifications API URL.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	private $api_url = '';
+
+	/**
+	 * Notification slug for dismissible content.
+	 *
+	 * @var string
+	 */
+	protected string $slug = '';
+
+	/**
+	 * Boot the in-App Notifications
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function boot(): void {
+		$this->api_url = $this->get_api_url();
+		$this->slugs   = $this->get_plugins();
+
+		/**
+		 * Allow plugins to hook in and add themselves,
+		 * running their own actions once In-App Notifications is initiated.
+		 *
+		 * @since TBD
+		 *
+		 * @param self $ian The In-App Notifications instance.
+		 */
+		do_action( 'tec_common_ian_preload', $this );
+	}
+
+	/**
+	 * Initializes the plugins and triggers the "loaded" action.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function init(): void {
+		/**
+		 * Allow plugins to hook in and add themselves,
+		 * running their own actions once IAN is initiated.
+		 *
+		 * @since TBD
+		 *
+		 * @param self $ian The IAN instance.
+		 */
+		do_action( 'tec_common_ian_loaded', $this );
+	}
+
+	/**
+	 * Register the Admin assets for the In-App Notifications.
+	 *
+	 * @since  TBD
+	 *
+	 * @return void
+	 */
+	public function register_assets(): void {
+		tribe_assets(
+			Tribe__Main::instance(),
+			[
+				[ 'ian-client-css', 'ian-client.css' ],
+				[ 'ian-client-js', 'ian-client.js', [ 'jquery' ] ],
+			],
+			'admin_enqueue_scripts',
+			[
+				'conditionals' => [ $this, 'is_ian_page' ],
+				'in_footer'    => false,
+				'localize'     => [
+					'name' => 'commonIan',
+					'data' => [
+						'ajax_url' => admin_url( 'admin-ajax.php' ),
+						'nonce'    => wp_create_nonce( 'common_ian_nonce' ),
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Get the API URL for the In-App Notifications.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	public function get_api_url() {
+		$api = 'https://ian.stellarwp.com/feed/organization/brand/product.json';
+
```

Comment:
What about setting this from The Events Calendar ( and Event Tickets or any of the premium plugins ). In TEC it would also have a filter to be able to easily target the url for that plugin's feed.

Reply:
Yes, it could [go here](https://github.com/the-events-calendar/the-events-calendar/pull/4798/files#diff-def87422643db7c06d83ff1115ce16c359b9d15d3f9155527c7fcc7d1650be6aR29), where we also have a function to decide which screen we are showing the notifications in. I am waiting for some decisions from @bordoni for that.

Thanks @jesseeproductions .